boxoffice/models/line_item.py

```diff
@@ -222,49 +223,72 @@ def counts_per_date_per_item(item_collection, user_tz):
     for item in item_collection.items:
         item_id = unicode(item.id)
         item_results = db.session.query('date', 'count').from_statement(
-            '''SELECT DATE_TRUNC('day', line_item.ordered_at AT TIME ZONE 'UTC' AT TIME ZONE :timezone)::date as date, count(line_item.id)
-            from line_item where item_id = :item_id and status = :status group by date order by date asc'''
-        ).params(timezone=user_tz, status=LINE_ITEM_STATUS.CONFIRMED, item_id=item.id)
-        for res in item_results:
-            if not date_item_counts.get(res[0].isoformat()):
+            '''SELECT DATE_TRUNC('day', line_item.ordered_at AT TIME ZONE 'UTC' AT TIME ZONE :timezone)::date as date, count(line_item.id) AS count
+            FROM line_item WHERE item_id = :item_id AND status = :status
+            GROUP BY date ORDER BY date ASC'''
+        ).params(timezone=user_tz, status=LINE_ITEM_STATUS.CONFIRMED, item_id=item.id).all()
+        for date_count in item_results:
+            if not date_item_counts.get(date_count.date):
                 # if this date hasn't been been mapped in date_item_counts yet
-                date_item_counts[res[0].isoformat()] = {item_id: res[1]}
+                date_item_counts[date_count.date] = {item_id: date_count.count}
             else:
                 # if it has been mapped, assign the count
-                date_item_counts[res[0].isoformat()][item_id] = res[1]
+                date_item_counts[date_count.date][item_id] = date_count.count
     return date_item_counts
 
 
-def sales_by_date(dates, user_tz, item_ids):
+def sales_by_date(sales_datetime, item_ids, timezone):
     """
-    Returns the net sales of line items sold on a date.
-    Accepts a list of dates.
-    ['2016-01-01', '2016-01-02'] => {'2016-01-01': }
+    Returns the sales amount accrued during the day for a given date/datetime,
+    list of item_ids and timezone.
     """
     if not item_ids:
         return None
 
-    date_sales = {}
+    start_at = midnight_to_utc(sales_datetime, timezone=timezone)
+    end_at = midnight_to_utc(sales_datetime + datetime.timedelta(days=1), timezone=timezone)
+    sales_on_date = db.session.query('sum').from_statement('''SELECT SUM(final_amount) FROM line_item
+        WHERE status=:status AND ordered_at >= :start_at AND ordered_at < :end_at
+        AND line_item.item_id IN :item_ids
+        ''').params(status=LINE_ITEM_STATUS.CONFIRMED, start_at=start_at, end_at=end_at, item_ids=tuple(item_ids)).scalar()
+    return sales_on_date if sales_on_date else Decimal(0)
```

Comment:
I tried this finding this exact conversation, but I think it was either for a variable assignment or a return expression.